### PR TITLE
Compile on aarch64

### PR DESCRIPTION
**Pull request by yisonPylkita**
With this Obliteration will compile on Apple M1. 

I'll try to setup aarch64 CI

### CODE REVIEW

**Comment by ultimaweapon** [comment]
Which CI you are going to use? AFAIK Apple M1 is not available on GitHub Actions yet.

**Comment by yisonPylkita** [comment]
I don't know yet. GitHub doesn't support arm for cloud yet (you can run on self-hosted runner though). Travis, CircleCI, Oracle CI, and others seems to require credit cards. 

I have a Raspberry Pi 4 laying around but I it could be super slow. I'll come back with a different PR when I have something. For now lets mere this if you don't have objections

File: src/kernel/src/process/module/recompiler/aarch64.rs [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+use iced_x86::{Instruction, Register};
+use std::collections::{HashMap, VecDeque};
+
+pub struct Aarch64Emitter<'input> {
+    proc: *mut Process,
+    input: &'input [u8],
+    segments: Vec<Segment>,
+    assembler: CodeAssembler,
+    jobs: VecDeque<(u64, usize, CodeLabel)>,
+    labels: HashMap<u64, CodeLabel>,
```
Comment (bayedieng) [inline src/kernel/src/process/module/recompiler/aarch64.rs:14]: These fields should be removed as they're specific to x86_64 instructions. We won't be using the same assembler.

File: src/kernel/src/process/module/recompiler/aarch64.rs [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+
+impl<'input> Aarch64Emitter<'input> {
+    pub fn new(proc: *mut Process, input: &'input [u8], segments: Vec<Segment>) -> Self {
+        Self {
+            proc,
+            input,
+            segments,
+            assembler: CodeAssembler::new(64).unwrap(),
+            jobs: VecDeque::new(),
+            labels: HashMap::new(),
```
Comment (bayedieng) [inline src/kernel/src/process/module/recompiler/aarch64.rs:26]: To be removed as well

**Comment by ultimaweapon** [comment]
It should be possible to cross compile for M1 from Intel, I guess. Let @bayedieng review this PR.

File: src/kernel/src/process/module/recompiler/aarch64.rs [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+            output_size: 0,
+        }
+    }
+
+    fn temp_register64(keep: Register) -> Register {
+        unimplemented!();
+    }
+}
```
Comment (bayedieng) [inline src/kernel/src/process/module/recompiler/aarch64.rs:34]: Also to be removed.

**Review by bayedieng — CHANGES_REQUESTED** [review]

**Comment by bayedieng** [comment]
> It should be possible to cross compile for M1 from Intel, I guess. Let @bayedieng review this PR.

Yes it is possible to do so by running ```rustup target add x86_64-apple-darwin``` in the CLI. It should be possible to run a game in theory using Rosetta 2 as well but have not tried this.

**Review by bayedieng — APPROVED** [review]

**Pull request merged**